### PR TITLE
A simple unit test to demonstrate how to use RTD

### PR DESCRIPTION
**Pull request by D-Ne0**
The concept behind RTD unit testing framework is to test app code in a non-meteor context environment. This means Meteor objects, it's packages etc. are not available during testing. So, while I was testing a template function, there were alot of errors like 'undefined is not an object "Template.server_channels"'. To remove these error I tried following techniques:
1. As suggested by RTD author: Create stubs for whatever you require in unit testing. Refer to this for more details: https://github.com/xolvio/rtd/issues/87. But this was not feasible because I had to create too many stubs.
2. Then I looked at test/karma.conf.js(originally present in dir node_modules/rtd). It has a configuration variable called 'files' which allows you to load only specific app files, thus removing the pain of creating any extra stubs. You can find in this pull request that I am loading only one app file(from test/karma.conf.js) which I had tested(test/client/chat/chat_connections/channels.js) and created stubs for only that file.

Conclusion: This is the best approach I could think of to unit test Waartaa. Since alot of code has already been written in Waartaa and RTD follows TDD techinque. So I think the only way to unit test is to load only those files(by specifying it in test/karma.conf.js) which you want to unit test.

Let me know If this approach suits you.

PS: Please also look at the directory structure inside test/unit/client where the first test is placed.

### CODE REVIEW

**Comment by D-Ne0** [comment]
Oops clicked the submit button soon!

To install RTD just run:

```
npm install
```

since I have already updated the package.json file.

File: app/client/chat/chat_connections/channels.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ Template.server_channels.channels = function (server_id) {
 }
 
 function serverChannelsCreatedCallback () {
-  $('#chat-servers .nano').nanoScroller();
+  $('#chat-servers .nan').nanoScroller();
```
Comment (rtnpro) [inline app/client/chat/chat_connections/channels.js:19]: why `nano` to `nan`?

File: test/unit/client/chat/chat_connections/channels.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  Template.stub('server_channels');
+  Template.stub('server_channel_item');
+  Template.stub('channel_menu');
+  Template.stub('edit_server_channel');
+
+  describe("Template.server_channels.channels", function () {
+    
+    it("asks for channels to be in alphabetical order, to not have last_updated field, to be active", function () {
+      var someLocaliCollectionCursor = {};
```
Comment (rtnpro) [inline test/unit/client/chat/chat_connections/channels.js]: Typo? shouldn't it be `someLocalCollectionCursor`?

File: app/client/chat/chat_connections/channels.js (hunk 1 above)
Comment (D-Ne0) [inline app/client/chat/chat_connections/channels.js:19]: Sorry, that was a mistake.

File: test/unit/client/chat/chat_connections/channels.js (hunk 2 above)
Comment (D-Ne0) [inline test/unit/client/chat/chat_connections/channels.js]: Yes.

**Comment by rtnpro** [comment]
@D-Ne0 Can you please rebase this branch on top of latest master? I want to try it out. Please let me know after rebasing it.

**Comment by D-Ne0** [comment]
@rtnpro Done. And I hope I have rebased it correctly.

**Comment by rtnpro** [comment]
@D-Ne0 

I have merged your code here: https://github.com/waartaa/waartaa/tree/tests.

How do I run the tests?

I am getting the following error:

```
./node_modules/rtd/rtd --debug
Running "bgShell:killAll" (bgShell) task
[D] Task source: /home/vagrant/waartaa/node_modules/rtd/node_modules/grunt-bg-shell/tasks/bg-shell.coffee

Running "downloadAndOrStartSelenium" task
[D] Task source: /home/vagrant/waartaa/node_modules/rtd/Gruntfile.js
Fatal error: spawn ENOENT
[vagrant@localhost waartaa]$ ./node_modules/rtd/rtd
Fatal error: spawn ENOENT
```

**Comment by D-Ne0** [comment]
This might help: http://johnattebury.com/blog/2013/07/grunt-js-error

**Comment by rtnpro** [comment]
Merged.

**Pull request closed without merging**